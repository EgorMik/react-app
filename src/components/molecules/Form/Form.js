import React from 'react';
import Input from '../../atoms/Input/Input'
import Button from '../../atoms/Button/Button'
import PropTypes from 'prop-types';

const Form = ({ value, updateText, handleAction, radio, handleRadio }) => {
    return (
        <div className="form">
          <form onSubmit={e => {e.preventDefault();}}>
              <Input
               radio={radio}
               value={value}
               updateText={updateText}
               handleRadio={handleRadio}
             />
              <Button 
              handleAction={handleAction}
              />
          </form>
        </div>
    );
  };
  Form.propTypes = {
    value: PropTypes.string,
    updateText: PropTypes.func,
    handleAction: PropTypes.func,
    radio: PropTypes.bool,
    handleRadio: PropTypes.func,
  }
  export default Form;
