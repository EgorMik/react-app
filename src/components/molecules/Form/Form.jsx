import Input from '../../atoms/Input/Input'
import Button from '../../atoms/Button/Button'

const Form = ({ value, updateText, handleAction, radio, handleRadio,onChangeDisplay }) => {
    return (
        <div className="inner">
          <form onSubmit={e => {e.preventDefault();}}>
              <Input
               radio={radio}
               value={value}
               updateText={updateText}
               handleRadio={handleRadio}
             />
              <Button 
              handleAction={handleAction}
              />
          </form>
        </div>
    );
  };
  export default Form;
