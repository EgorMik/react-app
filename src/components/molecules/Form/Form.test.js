import React from "react";
import renderer from "react-test-renderer";
import Form from "./Form";

describe("Form", () => {
  it("should renders correctly", () => {
    const tree = renderer.create(<Form name="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});