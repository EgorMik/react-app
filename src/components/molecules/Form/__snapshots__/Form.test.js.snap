// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should renders correctly 1`] = `
<div
  className="form"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="header"
    >
      <input
        className="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <input
        className="input"
        onChange={[Function]}
        placeholder="Add your important task"
        type="text"
      />
    </div>
    <button
      className="button"
    >
      Add todo
    </button>
  </form>
</div>
`;
